### ISSUE TITLE
Failing to verify the Catalog API server is running

### ISSUE THREAD

**Issue by mrbetta** [openshift/openshift-ansible#12015]
#### Description

> On a fresh install of Openshift 3.11 on virtual Centos7 the install process fails at 'FAILED - RETRYING: Verify that the Catalog API Server is running (1 retries left)." With what seems to be a name resolution issue.

##### Version

```
[openshift@okdmaster playbooks]$ ansible --version
ansible 2.8.5

[openshift@okdmaster playbooks]$ git describe
openshift-ansible-3.11.154-2-5-g4786a81
```

##### Steps To Reproduce
1. Followed the documentation to set up the environment. Installed Centos 7 minimal install. Created user account and pushed keys to all hosts.
2. Cloned the latest version of the repo.
3. Used the example template as a guide for a single master, single infrastructure, three host architecture.
4. ran ansible-playbook prerequisites.yaml with no error
5. ran deploy-cluster with error Verify API 

##### Expected Results
Describe what you expected to happen.
```
Install to complete successfully
```

##### Observed Results
Describe what is actually happening.

```
FAILED - RETRYING: Verify that the Catalog API Server is running (8 retries left).
FAILED - RETRYING: Verify that the Catalog API Server is running (7 retries left).
FAILED - RETRYING: Verify that the Catalog API Server is running (6 retries left).
FAILED - RETRYING: Verify that the Catalog API Server is running (5 retries left).
FAILED - RETRYING: Verify that the Catalog API Server is running (4 retries left).
FAILED - RETRYING: Verify that the Catalog API Server is running (3 retries left).
FAILED - RETRYING: Verify that the Catalog API Server is running (2 retries left).
FAILED - RETRYING: Verify that the Catalog API Server is running (1 retries left).
fatal: [okdmaster.DOMAIN.home]: FAILED! => {"attempts": 60, "changed": false, "cmd": ["curl", "-k", "https://apiserver.kube-service-catalog.svc/healthz"], "delta": "0:00:00.012505", "end": "2019-11-14 21:00:28.924834", "msg": "non-zero return code", "rc": 6, "start": "2019-11-14 21:00:28.912329", "stderr": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: apiserver.kube-service-catalog.svc; Unknown error", "stderr_lines": ["  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current", "                                 Dload  Upload   Total   Spent    Left  Speed", "", "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: apiserver.kube-service-catalog.svc; Unknown error"], "stdout": "", "stdout_lines": []}
...ignoring
```

For long output or logs, consider using a [gist](https://gist.github.com/)

##### Additional Information

Provide any additional information which may help us diagnose the
issue.

* Your operating system and version, ie: RHEL 7.2, Fedora 23 (`$ cat /etc/redhat-release`)
* Your inventory file (especially any non-standard configuration parameters)
* Sample code, etc

```
Centos 7.7 minimal install. Non-atomic.

All hosts are in DNS and can ping each other.

I've tried the solutions others reported without success including:
Removing the references to 127.0.0.1 in the /etc/hosts file
Different modifications in the /etc/dnsmasq.conf file after it's installed
Calling out the hostnames specifically in the inventory file
openshift_master_cluster_hostname=okdmaster.DOMAIN.home
openshift_master_cluster_public_hostname=okdmaster.DOMAIN.home

Inventory file
# Create an OSEv3 group that contains the masters, nodes, and etcd groups
[OSEv3:children]
masters
nodes
etcd

# Set variables common for all OSEv3 hosts
[OSEv3:vars]
# SSH user, this user should allow ssh based auth without requiring a password
ansible_ssh_user=openshift

# If ansible_ssh_user is not root, ansible_become must be set to true
ansible_become=true

openshift_deployment_type=origin
openshift_release=3.11
openshift_disable_check=memory_availability,disk_availability
openshift_master_cluster_hostname=okdmaster.DOMAIN.home
openshift_master_cluster_public_hostname=okdmaster.DOMAIN.home
# uncomment the following to enable htpasswd authentication; defaults to DenyAllPasswordIdentityProvider
#openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider'}]

# host group for masters
[masters]
okdmaster.DOMAIN.home

# host group for etcd
[etcd]
okdmaster.DOMAIN.home

# host group for nodes, includes region info
[nodes]
okdmaster.DOMAIN.home openshift_node_group_name='node-config-master'
okd1.DOMAIN.home openshift_node_group_name='node-config-compute'
okd2.DOMAIN.home openshift_node_group_name='node-config-compute'
okd3.DOMAIN.home openshift_node_group_name='node-config-compute'
okdinfra.DOMAIN.home openshift_node_group_name='node-config-infra'
```

**Comment by DanielChuDC** [comment]
same here. Any solution or workaround?

**Comment by mrbetta** [comment]
Not yet. I've noticed that if I ping the name it's trying to use it resolves and pings, but if I use host or nslookup it doesn't. I can't seem to find how it's resolving the name.

Are you using a Windows DNS server for your zone? There's been some issues reported about issues with wildcard records.

**Comment by DanielChuDC** [comment]
Hi, I think I resolved.

First downgrade your ansible version to 2.6

Try install again.

Then make sure your dnsmasq and skydns inside OKD is working. 
You can check by doing 'systemctl status dnsmasq'

Hope it helps

**Comment by mrbetta** [comment]
No luck for me. Centos 7 didn't have Ansible 2.6 in the repos, but I found an RPM and installed 2.6.14. Reverted the snapshots to my virtual machines and ran the pre-requisites playbook, then deploy-cluster again. It failed again in the same place. Catalog API server failed to verify after 60 retries.

**Comment by mrbetta** [comment]
Ok, I think it's working now. Not sure if this will help anyone else, since it seemed pretty specific to my environment.

My default DNS server resolves public IPs for the internet, and has a forward back to my 'Internal' domain for local resolution. When it was doing the Catalog Verify it didn't look like it was following the forward. When I changed the DNS server in /etc/dnsmasq.d/origin-upstream-dns.conf to point directly to the internal zone DNS server and restarted dnsmasq, it worked! I had to change it a little ways into the install process as one of the earlier steps of the install reverted it back to the default.
